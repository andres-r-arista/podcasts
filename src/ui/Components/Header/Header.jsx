import React from "react";
import { useNavigate } from "react-router-dom";

export const Header = () => {
  const navigate = useNavigate();
  return (
    <header className="p-2 border-b mb-2">
      <h1
        className="font-medium text-sky-800 hover:underline cursor-pointer"
        onClick={() => navigate(`/`)}
        data-testid="title"
      >
        PodCaster
      </h1>
    </header>
  );
};
