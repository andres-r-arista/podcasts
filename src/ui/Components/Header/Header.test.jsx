import { render, screen, fireEvent } from "@testing-library/react";

import { Header } from "./Header";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Tests in <Header />", () => {
  beforeEach(() => {
    mockedUsedNavigate.mockReset();
  });

  test("must match the snapshot", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test("must have a title that says 'PodCaster'", () => {
    render(<Header />);
    const title = screen.getByTestId("title");
    expect(title).toHaveTextContent("PodCaster");
  });

  test("Clicking on the title navigates to '/'", async () => {
    render(<Header />);
    const title = screen.getByTestId("title");
    await expect(mockedUsedNavigate).toHaveBeenCalledTimes(0);
    fireEvent.click(title);
    await expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    await expect(mockedUsedNavigate).toHaveBeenCalledWith("/");
  });
});
