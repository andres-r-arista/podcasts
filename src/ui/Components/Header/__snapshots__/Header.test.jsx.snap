// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Header /> must match the snapshot 1`] = `
<div>
  <header
    class="p-2 border-b md:mb-2"
  >
    <h1
      class="font-medium text-sky-800 hover:underline cursor-pointer"
      data-testid="title"
    >
      PodCaster
    </h1>
  </header>
</div>
`;
